### ISSUE TITLE
Explicitly call out in the exception message when client evaluation is due to an un-mapped property

### ISSUE THREAD

**Issue by ole1986** [dotnet/efcore#20483]
### Steps to reproduce

```C#
ctx.Person.Where(p => p.FirstName.Contains(a) || p.LastName.Contains(a))
```

**Exception:**

System.InvalidOperationException
  HResult=0x80131509
  Message=The LINQ expression 'DbSet<Person>
    .Where(s => s.FirstName.Contains(__filter_0) || s.LastName.Contains(__filter_0))' could not be translated. Either rewrite the query in a form that can be translated, or switch to client evaluation explicitly by inserting a call to either AsEnumerable(), AsAsyncEnumerable(), ToList(), or ToListAsync(). See https://go.microsoft.com/fwlink/?linkid=2101038 for more information.
  Source=Microsoft.EntityFrameworkCore

**Side note:**
Even the hint switching it to client evaluation does not seem to make sense IMO.
Furthermore is this an essential feature IMO

### Further technical details

EF Core version: 3.1
Database provider: Microsoft.EntityFrameworkCore.SqlServer
Target framework: .NET Core 3.1
Operating system: Windows 10
IDE: Visual Studio 2019 16.4.6

**Comment by ole1986** [comment]
Reason for this was that LastName property has been marked as `[NotMapped]`

**Comment by ajcvickers** [comment]
Re-opening to consider throwing a better exception for client eval due to un-mapped property, since this seems like a common cause.

**Comment by maumar** [comment]
fixed in 9e0f05b554d327646b0cb515ad3cd5d739766094
